larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/dto/TextPIIEntity.java

```diff
@@ -0,0 +1,88 @@
+package com.verygood.security.larky.modules.vgs.cerebro.text.analyzer.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import net.starlark.java.annot.StarlarkBuiltin;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.StarlarkFloat;
+import net.starlark.java.eval.StarlarkInt;
+import net.starlark.java.eval.StarlarkValue;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Builder
+@AllArgsConstructor
+@StarlarkBuiltin(
+    name = "TextPIIEntity",
+    category = "BUILTIN",
+    doc = "The type represents TextPIIAnalyzer text analysis result, and describes an instance of PII entity.")
+public class TextPIIEntity implements StarlarkValue {
+
+  // Type of the PII entity. e.g.: CARD_NUMBER, CRYPTO, etc.
+  private final String entityType;
+
+  // The PII detection score
+  private final StarlarkFloat score;
+
+  // Where the PII starts
+  private final StarlarkInt start;
+
+  // Where the PII ends
+  private final StarlarkInt end;
+
+  public static TextPIIEntity of(String entityType, double score, int start, int end) {
+    return TextPIIEntity.builder()
+        .entityType(entityType)
+        .score(StarlarkFloat.of(score))
+        .start(StarlarkInt.of(start))
+        .end(StarlarkInt.of(end))
+        .build();
+  }
+
+  @StarlarkMethod(
+      name = "entity_type",
+      structField = true,
+      doc = "PII entity entity type")
+  public String entityType() {
+    return entityType;
+  }
+
+  @StarlarkMethod(
+      name = "score",
+      structField = true,
+      doc = "The PII detection score")
+  public StarlarkFloat score() {
+    return score;
+  }
+
+  @StarlarkMethod(
+      name = "start",
+      structField = true,
+      doc = "Where the PII starts")
+  public StarlarkInt start() {
+    return start;
+  }
+
+  @StarlarkMethod(
+      name = "end",
+      structField = true,
+      doc = "Where the PII ends")
+  public StarlarkInt end() {
+    return end;
+  }
+
+  @StarlarkMethod(
```

Comment:
Why do we need a to_dict? should we inherit from a mutablestruct or larkyobject to get all the nice reprs and various utilities ?

Reply:
For data representation in base representation and allowing for flexible manipulations. The objects you've listed are package-private for types package.

Reply:
Resolved by https://github.com/verygoodsecurity/starlarky/pull/156/commits/2b2ad1b228e47bbffdf9ad30ee51a9d5db63ae72